docs/API_SPECS_DEVICE_MODELS.md

```diff
@@ -110,3 +116,373 @@ Content-Type: application/JSON
+	  "MessageSchema": {
+		"Name": "Chiller;v1",
+		"Format": "JSON",
+		"Fields": {
+		  "temperature": "Double",
+		  "temperature_unit": "Text",
+		  "humidity": "Double",
+		  "humidity_unit": "Text",
+		  "pressure": "Double",
+		  "pressure_unit": "Text"
+		}
+	  }
+	}
+  ],
+  "CloudToDeviceMethods": {},
+  "$metadata": {
+    "$type": "DeviceModel;2",
+    "$uri": "/v1/devicemodels/12345678-c6c0-49ae-9bfa-123456789012",
+    "$created": "2018-03-29T23:33:03+00:00",
+    "$modified": "2018-03-29T23:33:03+00:00"
+  }
+}
+```
+
+## Modifying a device model
+
+### Modifying stock device models
+
+Directly modify the JSON configuration files stored in the file system.
+
+[Device model docs](https://docs.microsoft.com/azure/iot-suite/iot-suite-remote-monitoring-test#define-the-characteristics-of-the-new-device-type)
+
+### Modifying custom device models
+
+Custom device models can be modified by calling PUT and passing the existing 
+device model ETag and ID. The PUT request should be preceded by a GET request,
+to pull the existing device model content, before editing it and sending the
+modification(s).
+
+If a client sends a PUT request with a non-existent ID, the service automatically
+creates a new model and stores it in the database.
+
+Request:
+```
+PUT /v1/devicemodels/${id}
+Content-Type: application/json; charset=utf-8
+```
+```json
+{
+  "ETag": "12345678-abcd-1c4a-fa9b-123456789012",
+  "Id": "12345678-c6c0-49ae-9bfa-123456789012",
+  "Version": "0.0.1",
+  "Name": "Chiller",
+  "Description": "Chiller with external temperature, humidity and pressure sensors.",
+  "Protocol": "MQTT",
+  "Type": "custom",
+  "Simulation": {
+    "InitialState": {
+      "online": true,
+      "temperature": 75,
+      "temperature_unit": "F",
+      "humidity": 70,
+      "humidity_unit": "%",
+      "pressure": 150,
+      "pressure_unit": "psig",
+      "simulation_state": "normal_pressure"
+    },
+    "Interval": "00:00:10",
+    "Scripts": [
+      {
+        "Type": "javascript",
+        "Path": "chiller-01-state.js"
+      }
+    ]
+  },
+  "Properties": {},
+  "Telemetry": [
+    {
+      "Interval": "00:00:10",
+      "MessageTemplate": "{\"temperature\":${temperature},\"temperature_unit\":\"${temperature_unit}\",\"humidity\":${humidity},\"humidity_unit\":\"${humidity_unit}\",\"pressure\":${pressure},\"pressure_unit\":\"${pressure_unit}\"}",
+      "MessageSchema": {
+        "Name": "chiller-sensors;v1",
+        "Format": "JSON",
+        "Fields": {
+          "temperature": "Double",
+          "temperature_unit": "Text",
+          "humidity": "Double",
+          "humidity_unit": "Text",
+          "pressure": "Double",
+          "pressure_unit": "Text"
+        }
+      }
+    }
+  ],
+  "CloudToDeviceMethods": {},
+}
+```
+
+Response:
+```
+200 OK
+Content-Type: application/json; charset=utf-8
+```
+```json
+{
+  "ETag": "00000000-000000-0000-0000-000000",
```

Comment:
fix ETag and Id, also below in the metadata

Reply:
Done.